### PR TITLE
update: Fixed returned error codes

### PR DESCRIPTION
**Pull request by vargac**
- returns NO_ERR when succes
- changed example (new checking for errors)

### CODE REVIEW

File: NXTViSy.nxc [hunk 1]
```diff
@@ -11,26 +11,27 @@ void SetSensorNXTViSy(byte port)
...
+    byte inBuffer[2] = { I2C_ADDRESS, GET_DATA };
     byte outBuffer[];
+    long error;
     
     while (I2CCheckStatus(NXTViSy_port) != NO_ERR)
         ;
-    if (I2CBytes(NXTViSy_port, inBuffer, received_bytes_count, outBuffer) != NO_ERR)
-        return false;
+    if (I2CBytes(NXTViSy_port, inBuffer, received_bytes_count, outBuffer) == false)
+        return 0x01;
```
Comment (mrshu) [inline NXTViSy.nxc]: @vargac if we want to have different return values that would signalize different errors or success they should be defined as constants. This way it is very hard to interpret what `0x01` actually means.

File: NXTViSy.nxc [hunk 2]
```diff
@@ -11,26 +11,27 @@ void SetSensorNXTViSy(byte port)
...
     if (received_bytes_count != 0x03)
-        return false;
+        return 0x02;
 
     distance = outBuffer[0];
     distance |= outBuffer[1] << 8;
     action = outBuffer[2];
     
-    return true;
+    return 0x00;
```
Comment (mrshu) [inline NXTViSy.nxc]: @vargac Kudos for using `0` for success!

File: example.nxc [hunk 3]
```diff
@@ -6,18 +6,28 @@
 task main()
 {
     SetSensorNXTViSy(PORT);
-    NXTViSyInit(SIGNATURE, 10, 24 * 100);
+    long err = NXTViSyInit(SIGNATURE, 10, 24 * 100);
+    if (err != NO_ERR) {
+        NumOut(0, LCD_LINE4, err);
```
Comment (mrshu) [inline example.nxc]: @vargac There should probably be also some textual information on the LCD screen which would help trace the problem to this line.

Something like `TextOut(0, LCD_LINE3, "Error: NXTViSyInit")` perhaps?

File: example.nxc [hunk 4]
```diff
@@ -6,18 +6,28 @@
...
     int distance;
     byte action;
     string writable_action = "x";
 
     while (true) {
-        NXTViSyGetData(distance, action);
+        ClearScreen();
+        err = NXTViSyGetData(distance, action);
+        if (err != NO_ERR) {
+            NumOut(0, LCD_LINE5, err);
```
Comment (mrshu) [inline example.nxc]: @vargac same sa above, though I understand your intention with different lines for different errors.

File: example.nxc [hunk 5]
```diff
@@ -6,18 +6,28 @@
 task main()
 {
     SetSensorNXTViSy(PORT);
-    NXTViSyInit(SIGNATURE, 10, 24 * 100);
+    long err = NXTViSyInit(SIGNATURE, 10, 24 * 100);
+    if (err != NO_ERR) {
+        NumOut(0, LCD_LINE4, err);
+        while(1);
```
Comment (mrshu) [inline example.nxc]: @vargac again, I'd very much prefer to see the semicolon (`;`) on its own line. If for no other reason, then just to make @burlito happy.

**Comment by mrshu** [comment]
@vargac Overall this looks quite good, just please adress those few comments of mine (they are really mostly nitpicks).

File: example.nxc [hunk 6]
```diff
@@ -6,18 +6,41 @@
...
     int distance;
     byte action;
     string writable_action = "x";
 
     while (true) {
-        NXTViSyGetData(distance, action);
+        ClearScreen();
+        err = NXTViSyGetData(distance, action);
+        if (err == GET_DATA_I2C_BYTES_ERR) {
+            TextOut(0, LCD_LINE5, "I2CBytes error in function");
```
Comment (mrshu) [inline example.nxc]: @vargac could you please refactor these 5 lines into a function that would take two arguments and do precisely what these 5 lines do (that is print out those two lines and wait forever).

File: example.nxc [hunk 7]
```diff
@@ -6,18 +6,41 @@
...
+            TextOut(0, LCD_LINE6, "GetData()");
+            while (true) {
+                ;
+            }
+        }
+        if (err == GET_DATA_RECEIVED_BYTES_ERR) {
+            TextOut(0, LCD_LINE5, "Unexpected count of bytes");
```
Comment (mrshu) [inline example.nxc]: @vargac we could use the function mentioned above here.

**Comment by mrshu** [comment]
@vargac sorry, I kind of forgot about this.

It generally looks very nice -- I only have a few comments.

**Comment by vargac** [comment]
@mrshu Please merge this so we have running code in master branch.

**Comment by mrshu** [comment]
@vargac Ok, looks pretty good now, sorry that it took this long

**Pull request merged**